systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java

```diff
@@ -90,12 +199,76 @@ public static DeploymentBuilder defaultAdminClient(String namespaceName, String
             .endSpec();
     }
 
-    private static String getAdminClientScramConfig(String namespaceName, String userName) {
-        final String saslJaasConfigEncrypted = kubeClient().getSecret(namespaceName, userName).getData().get("sasl.jaas.config");
+    ///////////////////////////////////////////
+    //   Admin Client Configuration and envs
+    ///////////////////////////////////////////
+
+    /**
+     * Constructs the (SASL) SCRAM configuration string from a secret for an admin client based on the user and security protocol.
+     * Works for SASL-PLAIN and SASL-SSL alike.
+     *
+     * @param namespace the namespace in which the secret is stored
+     * @param userName the name of the user (also used as the secret name) to fetch the SASL JAAS config
+     * @param securityProtocol the security protocol to use (either SASL_PLAINTEXT or SASL_SSL)
+     * @return a {@link String} containing the SASL mechanism, security protocol, and the SASL JAAS configuration
+     */
+    private static String getAdminClientScramConfig(String namespace, String userName, SecurityProtocol securityProtocol) {
+        final String saslJaasConfigEncrypted = kubeClient().getSecret(namespace, userName).getData().get("sasl.jaas.config");
         final String saslJaasConfigDecrypted = Util.decodeFromBase64(saslJaasConfigEncrypted);
 
         return "sasl.mechanism=SCRAM-SHA-512\n" +
-            "security.protocol=" + SecurityProtocol.SASL_PLAINTEXT + "\n" +
+            "security.protocol=" + securityProtocol + "\n" +
             "sasl.jaas.config=" + saslJaasConfigDecrypted;
     }
+
+    /**
+     * Creates an {@link EnvVar} environment variable for the Kafka cluster CA certificate, used for
+     * configuring Kafka clients to trust the cluster's CA, thereby enabling TlS and ScramSha over TlS communication.
+     *
+     * @param clusterName the name of the Kafka cluster, used to derive the name of the Kubernetes secret containing the CA certificate
+     * @return an {@link EnvVar} instance representing the CA certificate environment variable
+     */
+    private static EnvVar getClusterCaCertEnv(String clusterName) {
+        return  new EnvVarBuilder()
+            .withName("CA_CRT")
+            .withNewValueFrom()
+                .withNewSecretKeyRef()
+                    .withName(KafkaResources.clusterCaCertificateSecretName(clusterName))
+                    .withKey("ca.crt")
+                .endSecretKeyRef()
+            .endValueFrom()
+            .build();
+    }
```

Comment:
This seems like a generic to use in other resources also.

Reply:
there are only few (2) resources that needs something like this and 1 of them are clients which have this provided internally.